### PR TITLE
Document Repartition and ReduceStateByKey

### PR DESCRIPTION
**Pull request by xitep**
* Javadocs
* Some cleanup regarding generic type parameters on the `ReduceStateByKey` operator

### CODE REVIEW

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java [hunk 1]
```diff
@@ -32,26 +32,60 @@
...
+ *
+ * Note that defining a {@link Windowing} strategy is actually optional. If
+ * none is defined the operator will act in the so-call "attached windowing"
+ * mode. I.e. it will attach itself to the windowing strategy active on they
+ * way from its input.
+ *
+ * @param <IN>     the type of input elements
+ * @param <KEY>    the type of the key (result type of {@code #keyBy}
+ * @param <VALUE>  the type of the accumulated values (result type of {@code #valueBy})
+ * @param <OUT>    the type of the output elements (result type of the accumulated state)
```
Comment (vanekjar) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java]: Cool, like it :+1:

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java [hunk 2]
```diff
@@ -32,26 +32,60 @@
...
- * @param <WIN>    Type of input records being windowed
- * @param <KEY>    Output type of #keyBy method
- * @param <VALUE>  Output type of #valueBy method
- * @param <KEYOUT> Type of output key
- * @param <OUT>    Type of output value
+ * The operator assigns each input item to a set of windows (through a user
+ * provided {@link Windowing} implementation) and turns the item into a
+ * key/value pair. For each of the assigned windows the extracted value is
+ * accumulated using a user provided {@link StateFactory state implementation}
+ * under the extracted key. I.e. the value is accumulated into a state
```
Comment (vanekjar) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java]: Just a detail. I wouldn't start a new sentence with `I.e.`. It looks really weird.

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java [hunk 3]
```diff
@@ -113,6 +185,20 @@
       this.valueExtractor = Objects.requireNonNull(valueExtractor);
     }
 
+    /**
+     * Specifies a factory for creating new/empty/blank state instances.
+     *
+     * @param <OUT> the type of output elements the creates states will produce;
```
Comment (vanekjar) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java]: There must be some typo. Not sure what was the original intention.

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java [hunk 4]
```diff
@@ -91,6 +149,20 @@
...
     }
 
+    /**
+     * Specifies the function to derive a value from the
+     * {@link ReduceStateByKey operator's} input elements to get later
+     * accumulated by a later supplied state implementation.
+     *
+     * @param <VALUE> the type of the extracted values
+     *
+     * @param valueExtractor a user defined function to extra values from the
```
Comment (vanekjar) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceStateByKey.java]: "extract"

**Comment by xitep** [comment]
* many thanks, typos corrected, resolved conflicts and rebased; please have travis report all is fine before merging.
* once again, thanks for going through it! it's greatly appreciated 👍

**Pull request merged**